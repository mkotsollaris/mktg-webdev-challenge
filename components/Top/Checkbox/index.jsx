import { useState } from 'react'
import style from './style.module.css'

const Checkbox3 = ({
  checked,
  spanText = 'Hide people missing a profile image',
  ...props
}) => {
  // const [isChecked, setIsChecked] = useState();
  const [isFocused, setIsFocused] = useState(false)

  const classNames = isFocused ? ` ${style['on-focus']}` : null

  return (
    <div className={`${style.container}`}>
      <div>
        <div className={style.grid}>
          <input
            className={style.input}
            onFocus={() => {
              setIsFocused(true)
            }}
            onBlur={() => {
              setIsFocused(false)
            }}
            type="checkbox"
            checked={checked}
            {...props}
          />
          <span className={`${style.checkmark} ${classNames}`}></span>
          <span className={style['small-span']}>{spanText}</span>
        </div>
      </div>
    </div>
  )
}

export default Checkbox3
