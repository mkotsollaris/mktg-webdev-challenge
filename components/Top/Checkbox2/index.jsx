import React from 'react'
import styled from 'styled-components'

const CheckboxContainer = styled.div`
  display: inline-block;
  vertical-align: middle;
  width: 22px;
  height: 22px;
`

const Icon = styled.svg`
  fill: none;
  stroke: white;
  stroke-width: 2px;
`
// Hide checkbox visually but remain accessible to screen readers.
// Source: https://polished.js.org/docs/#hidevisually
const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  border: 0;
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`

const StyledCheckbox = styled.div`
  display: inline-block;
  width: 22px;
  height: 22px;
  background: ${(props) => (props.checked ? 'dodgerblue' : 'white')};
  border-radius: 3px;
  border: 1px solid dodgerblue;
  transition: all 150ms;

  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 2px dodgerblue;
  }

  ${Icon} {
    visibility: ${(props) => (props.checked ? 'visible' : 'hidden')};
  }
`

const Checkbox = ({ className, checked, ...props }) => {
  return (
    <CheckboxContainer className={className}>
      <HiddenCheckbox checked={checked} {...props} />
      <StyledCheckbox checked={checked}>
        <Icon viewBox="0 0 24 24">
          <polyline points="17 7 9 17 5 12" />
        </Icon>
      </StyledCheckbox>
    </CheckboxContainer>
  )
}

export default Checkbox
