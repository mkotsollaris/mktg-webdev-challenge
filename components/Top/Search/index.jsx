import { useContext } from 'react'
import style from './style.module.css'
import searchIconSrc from '../../../public/img/search.svg'
import AppContext from '../../../context/AppContext'

const Search = () => {
  const { setSearchValue } = useContext(AppContext)

  return (
    <>
      <div className={style['search-grid']}>
        <div className={style['search-icon']}>
          <img src={searchIconSrc} alt="search" />
        </div>
        <div>
          <input
            onChange={(e) => setSearchValue(e.target.value.toLowerCase())}
            type="text"
            placeholder="Search people by name"
          />
        </div>
      </div>
    </>
  )
}

export default Search
