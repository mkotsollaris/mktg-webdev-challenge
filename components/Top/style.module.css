.top {
  text-align: center;
  margin-top: 2rem;
  display: grid;
  justify-content: center;
  align-content: center;
  align-items: center;
  margin-bottom: 2rem;
}

.small-span {
  font-size: 0.9rem;
  font-weight: lighter;
  color: darkslategray;
}

.medium-span {
  font-size: 1.2rem;
  font-weight: lighter;
  color: darkslategray;
}

.margin-bottom-sm {
  margin-bottom: 1rem;
}

.large-span {
  font-size: 3rem;
  font-weight: 100;
  margin: 1rem;
}

.bottom-grid {
  display: grid;
  grid-template-columns: auto auto;
  grid-column-gap: 3%;
  justify-content: center;
  align-content: center;
  align-items: center;
}

.three-column-grid {
  grid-template-columns: auto auto auto;
}
