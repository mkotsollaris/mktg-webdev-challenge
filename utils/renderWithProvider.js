import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
