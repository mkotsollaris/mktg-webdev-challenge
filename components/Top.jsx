import Search from './Search'
import SubTitle from './SubTitle'
import Title from './Title'
import Checkbox from './CheckBox'
import style from './style.module.css'

const Top = () => {
  return (
    <div className={style.top}>
      <Title />
      <Search />
      <SubTitle />
      <div className={style['checkbox-grid']}>
        <Checkbox />
        <span>Hide people missing a profile image</span>
      </div>
    </div>
  )
}

export default Top
