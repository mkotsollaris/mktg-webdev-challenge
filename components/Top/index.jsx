import Search from './Search'
// import Checkbox from './Checkbox'
import style from './style.module.css'
import componentStyle from '../style.module.css'
import AppContext from '../../context/AppContext'
import { useContext } from 'react'
import ClearFilter from '../ClearFilter'
import { useMediaQuery } from '@material-ui/core'
import Checkbox2 from './Checkbox2'
const Top = () => {
  let isPageSmall = useMediaQuery('(max-width: 960px)')

  const {
    hideWithMissingProfileImg,
    setHideWithMissingProfileImg,
    filteredDepartments,
  } = useContext(AppContext)

  const checkBoxLabel = (
    <label>
      <Checkbox2
        checked={hideWithMissingProfileImg}
        onChange={() => {
          setHideWithMissingProfileImg(!hideWithMissingProfileImg)
        }}
      />
      <span className={style['small-span']} style={{ marginLeft: 8 }}>
        Hide people missing a profile image
      </span>
    </label>
  )

  const content =
    isPageSmall && filteredDepartments.length > 0 ? (
      <>
        <div
          className={`${style['bottom-grid']} ${style['three-column-grid']}`}
        >
          <div
            className={`${componentStyle.clickable} ${componentStyle['non-selectable']}`}
          >
            {checkBoxLabel}
          </div>
        </div>
        <div>
          <ClearFilter />
        </div>
      </>
    ) : (
      <div
        className={`${style['bottom-grid']} ${componentStyle.clickable} ${componentStyle['non-selectable']}`}
      >
        {checkBoxLabel}
      </div>
    )

  return (
    <div className={style.top}>
      <h1 className={style['large-span']}>Hashicorp Humans</h1>
      <span className={`${style['medium-span']} ${style['margin-bottom-sm']}`}>
        Find a HashiCorp human
      </span>
      <Search />
      {content}
    </div>
  )
}

export default Top
