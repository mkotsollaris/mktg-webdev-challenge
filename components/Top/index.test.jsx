import Top from '.'
import renderWithProvider from '../../utils/renderWithProvider'

it('renders', () => {
  const { asFragment } = renderWithProvider({}, Top)
  expect(asFragment()).toMatchSnapshot()
})
