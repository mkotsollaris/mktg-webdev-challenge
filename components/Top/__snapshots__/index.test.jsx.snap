// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hashicorp Humans
    </h1>
    <span
      class="undefined undefined"
    >
      Find a HashiCorp human
    </span>
    <div
      role="search"
    >
      <div>
        <img
          alt="search"
          src="[object Object]"
        />
      </div>
      <div>
        <input
          placeholder="Search people by name"
          type="search"
        />
      </div>
    </div>
    <div
      class="undefined undefined"
    >
      <label>
        <div
          class="undefined"
        >
          <div>
            <div>
              <input
                type="checkbox"
              />
              <span
                class="undefined null"
              />
              <span>
                Hide people missing a profile image
              </span>
            </div>
          </div>
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
